manifests/osfamily/solaris.pp

```diff
@@ -51,12 +51,23 @@
 
       $pkgrepo_dir = '/etc/puppetlabs/installer/solaris.repo'
 
+      # Make sure the pkg publishers are all available.  Broken
+      # publisher entries will stop the installation process.
+      # This must happen before removing any packages.
+      exec { 'puppet_agent ensure pkg publishers are available':
+        command   => 'pkg refresh',
```

Comment:
I am definitely not an expert on this but do you see any problems with us using `pkgrepo refresh -s ${pkgrepo_dir}` to refresh only our repo instead of refreshing all of the indexes in the repos on the system?